docs/advanced/api/vitest.md

```diff
@@ -0,0 +1,521 @@
+```
+
+This returns the root context object. This is a shorthand for `vitest.getRootProject().getProvidedContext`.
+
+## getProjectByName
+
+```ts
+function getProjectByName(name: string): TestProject
+```
+
+This method returns the project by its name. Simillar to calling `vitest.projects.find`.
+
+::: warning
+In case the project doesn't exist, this method will return the root project - make sure to check the names again if the project you are looking for is the one returned.
+
+If user didn't customize a name, the Vitest will assign an empty string as a name.
+:::
+
+## globTestSpecifications
+
+```ts
+function globTestSpecifications(
+  filters?: string[],
+): Promise<TestSpecification[]>
+```
+
+This method constructs new [test specifications](/advanced/api/test-specification) by collecting every test in all projects with [`project.globTestFiles`](/advanced/api/test-project#globtestfiles). It accepts string filters to match the test files - these are the same filters that [CLI supports](/guide/filtering#cli).
+
+This method automatically caches all test specifications. When you call [`getModuleSpecifications`](#getmodulespecifications) next time, it will return the same specifications unless [`clearSpecificationsCache`](#clearspecificationscache) was called before that.
+
+::: warning
+As of Vitest 3, it's possible to have multiple test specifications with the same module ID (file path) if `poolMatchGlob` has several pools or if `typecheck` is enabled. This possibility will be removed in Vitest 4.
+:::
+
+```ts
+const specifications = await vitest.globTestSpecifications(['my-filter'])
+// [TestSpecification{ moduleId: '/tests/my-filter.test.ts' }]
+console.log(specifications)
+```
+
+## getRelevantTestSpecifications
+
+```ts
+function getRelevantTestSpecifications(
+  filters?: string[]
+): Promise<TestSpecification[]>
+```
+
+This method resolves every test specification by calling [`project.globTestFiles`](/advanced/api/test-project#globtestfiles). It accepts string filters to match the test files - these are the same filters that [CLI supports](/guide/filtering#cli). If `--changed` flag was specified, the list will be filtered to include only files that changed. `getRelevantTestSpecifications` doesn't run any test files.
+
+::: warning
+This method can be slow because it needs to filter `--changed` flags. Do not use it if you just need a list of test files.
+
+- If you need to get the list of specifications for known test files, use [`getModuleSpecifications`](#getmodulespecifications) instead.
+- If you need to get the list of all possible test files, use [`globTestSpecifications`](#globtestspecifications).
+:::
+
+## mergeReports
+
+```ts
+function mergeReports(directory?: string): Promise<TestRunResult>
+```
+
+Merge reports from multiple runs located in the specified directory (`--merge-reports` if not specified). This value can also be set on `config.mergeReports` (by default, it will read `.vitest-reports` folder).
```

Comment:
> located in the specified directory (`--merge-reports` if not specified). 

`--merge-reports` is not directory.

Reply:
`--merge-reports` resolves into a directory if specified as `true`

Reply:
```
$ vitest --help
--merge-reports [path] Paths to blob reports directory. If this options is used, Vitest won\'t run any tests, it will only report previously recorded tests
```

Reply:
@AriPerkkio anything to add?

Reply:
No strong opinions but this change would read out better to me 😄 

```diff
- located in the specified directory (`--merge-reports` if not specified).
+ located in the specified directory (value from `--merge-reports` if not specified).
```